### PR TITLE
Addition of breadcrumb

### PR DESCRIPTION
**Pull request by nathancashmoredwp**

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const i18n = require('./middleware/i18n');
 const errorHandler = require('./middleware/error-handler');
 const healthCheckController = require('./controllers/health-check-controller');
 const helmet = require('helmet');
+const breadcrumbModel = require('./controllers/breadcrumb-view-model');
```
Comment (chilliwilly77) [inline app/app.js]: rename file to breadcrumb-model

File: app/app.js [hunk 2]
```diff
@@ -46,6 +47,7 @@ app.use((req, res, next) => {
       govukTemplate:
         '../../vendor/govuk_template_mustache_inheritance/views/layouts/govuk_template',
       googleTagManager: 'partials/google-tag-manager',
+      breadcrumbTemplate: 'partials/breadcrumb',
```
Comment (chilliwilly77) [inline app/app.js]: This is technically a partial. Maybe just rename to breadcrumb to fit in with existing partial naming convention?

**Comment by lukasz-kaniowski** [comment]
👍

File: app/controllers/breadcrumb-view-model.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+const i18n = require('i18n');
+const resources = require('../data/resources');
+
+/* eslint-disable no-underscore-dangle */
+
+const requestToBreadcrumbMap = [
+  { requestRegEx: '/search', page: 'search', trail: ['dashboard'] },
```
Comment (chilliwilly77) [inline app/controllers/breadcrumb-view-model.js]: lowercase regex. Also, remove quotations to make this a real regex

File: app/controllers/breadcrumb-view-model.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  return ({ link: `${request}`, title: i18n.__('breadcrumb.search') });
+}
+
+module.exports = function (request) {
+  const breadcrumb = [];
+  const breadcrumbRequest = requestToBreadcrumbMap.find(x => request.match(x.requestRegEx));
+
+  if (breadcrumbRequest) {
+    breadcrumbRequest.trail.forEach(crumb => {
+      breadcrumb.push(crumbDefaults[crumb]);
```
Comment (chilliwilly77) [inline app/controllers/breadcrumb-view-model.js]: I find this confusing because crumbDefaults is not an array.

File: app/locales/en.json [hunk 5]
```diff
@@ -44,5 +44,9 @@
     },
     "linkText": "Check it out",
     "copyJournalMessage": "Copy message to clipboard"
+  },
+  "breadcrumb": {
+    "dashboard": "Homepage",
+    "search": "Search"
   }
-}
+}
```
Comment (chilliwilly77) [inline app/locales/en.json]: new line

File: test/unit/breadcrumb-view-model.spec.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+require('../../app/middleware/i18n');
```
Comment (chilliwilly77) [inline test/unit/breadcrumb-view-model.spec.js]: rename this file to breadcrumb-view.spec.js too

**Pull request merged**